src/shared/components/query/GeneSetSelector.tsx

```diff
@@ -94,6 +97,9 @@ export default class GeneSetSelector extends QueryStoreComponent<GeneSetSelector
 
 				{!!(this.store.mutSigForSingleStudy.result.length || this.store.gisticForSingleStudy.result.length) && (
 					<FlexRow padded className={styles.buttonRow}>
+						<LoadingIndicator
+							isLoading={!!([this.store.mutSigForSingleStudy, this.store.gisticForSingleStudy].some(promise => promise.isPending))}
```

Comment:
We want to show loading indicator here instead of next to the `SectionHeader`, since we don't need the loader initially for this section. It is only loading when a study is selected.